src/gen_apple/ga_dev_xbee.cpp

```diff
@@ -1,26 +1,117 @@
 #include "ga_dev_xbee.h"
 
+struct XBee xbee;
+
 void ga_dev_xbee_open(void)
 {
     soft_serial.begin(9600);
-    xbee.begin(soft_serial);
+    xbee.serial = &soft_serial;
+    xbee.api_id = ZB_TX_REQUEST;
+    xbee.frame_id = DEFAULT_FRAME_ID;
+    xbee.broadcast_range = ZB_BROADCAST_RADIUS_MAX_HOPS;
+    xbee.option = ZB_TX_UNICAST;
+    xbee.addr16 = ZB_BROADCAST_PACKET;
+    xbee.msb = 0;
+    xbee.lsb = 0;
 }
 
-int ga_dev_xbee_avail(void)
-{
+int ga_dev_xbee_avail(void) {
     return Serial.available();
 }
 
-int ga_dev_xbee_read(void)
-{
+int ga_dev_xbee_read(void) {
     return Serial.read();
 }
 
-void ga_dev_xbee_write(uint8_t *data, int data_len)
-{
-    XBeeAddress64 addr64 = XBeeAddress64(0, 0);
-    ZBTxRequest zbtx = ZBTxRequest(addr64, data, data_len);
+void ga_dev_xbee_write(uint8_t val) {
+    xbee.serial -> write(val);
+}
+
+uint8_t ga_dev_xbee_frame_length(int data_len) {
+    return ZB_TX_API_LENGTH + data_len;
+}
+
+uint8_t ga_dev_xbee_frame_data(uint8_t* data, int pos) {
+    if (pos == 0) {
+		return (xbee.msb >> 24) & 0xff;
+	}
+    else if (pos == 1) {
+		return (xbee.msb >> 16) & 0xff;
+	}
+    else if (pos == 2) {
+		return (xbee.msb >> 8) & 0xff;
+	}
+    else if (pos == 3) {
+		return xbee.msb & 0xff;
+	}
+    else if (pos == 4) {
+		return (xbee.lsb >> 24) & 0xff;
+	}
+    else if (pos == 5) {
+		return  (xbee.lsb >> 16) & 0xff;
+	}
+    else if (pos == 6) {
+		return (xbee.lsb >> 8) & 0xff;
+	}
+    else if (pos == 7) {
+		return xbee.lsb & 0xff;
+	}
+    else if (pos == 8) {
+		return (xbee.addr16 >> 8) & 0xff;
+	}
+    else if (pos == 9) {
+		return xbee.addr16 & 0xff;
+	}
+    else if (pos == 10) {
+		return xbee.broadcast_range;
+	}
+    else if (pos == 11) {
+        return xbee.option;
+	}
+    else {
+		return data[pos - ZB_TX_API_LENGTH];
+	}
+}
+
+void ga_dev_xbee_send_byte(uint8_t b, bool escape) {
+    if (escape && (b == START_BYTE || b == ESCAPE || b == XON || b == XOFF)) {
+        ga_dev_xbee_write(ESCAPE);
+        ga_dev_xbee_write(b ^ 0x20);
+    }
 
-    xbee.send(zbtx);
+    else {
+        ga_dev_xbee_write(b);
+    }
 }
 
+void ga_dev_xbee_send(uint8_t* data, int data_len) {
+    ga_dev_xbee_send_byte(START_BYTE, false);
+
+	// send length
```

Comment:
There is a mixed use of tabs and spaces here; since all of the other files use spaces you should follow the current convention.